lib/FirehoseClient/index.js

```diff
@@ -1,77 +1,50 @@
 const SymMessageParser = require('../SymMessageParser')
 const request = require('../Request')
+const { agentRequest } = require('../Request/clients')
 
-var FirehoseClient = {}
+const FirehoseClient = {}
 
-FirehoseClient.registerBot = (token) => {
+FirehoseClient.registerBot = token => {
   FirehoseClient.psToken = token
 }
 
 // Create Firehose v1.0 using https://rest-api.symphony.com/v1.53/reference?showHidden=1a1dd#create-firehose-v4
-FirehoseClient.createFirehose = () => {
-  var options = {
-    'hostname': SymConfigLoader.SymConfig.agentHost,
-    'port': SymConfigLoader.SymConfig.agentPort,
-    'path': '/agent/v4/firehose/create',
-    'method': 'POST',
-    'headers': {
-      'sessionToken': SymBotAuth.sessionAuthToken,
-      'keyManagerToken': SymBotAuth.kmAuthToken
-    },
-    'agent': SymConfigLoader.SymConfig.agentProxy
-  }
-
-  return request(options, 'FirehoseClient/createFirehose')
-}
+FirehoseClient.createFirehose = () =>
+  agentRequest('post', '/agent/v4/firehose/create', 'FirehoseClient/createFirehose')
 
 // Read Firehose v1.0 using https://rest-api.symphony.com/v1.53/reference?showHidden=1a1dd#read-firehose-v4
-FirehoseClient.getEventsFromFirehose = (firehoseId) => {
-  var defer = Q.defer()
-
-  var options = {
-    'hostname': SymConfigLoader.SymConfig.agentHost,
-    'port': SymConfigLoader.SymConfig.agentPort,
-    'path': '/agent/v4/firehose/' + firehoseId + '/read',
-    'method': 'GET',
-    'headers': {
-      'sessionToken': SymBotAuth.sessionAuthToken,
-      'keyManagerToken': SymBotAuth.kmAuthToken
+FirehoseClient.getEventsFromFirehose = firehoseId => {
+  const options = {
+    hostname: SymConfigLoader.SymConfig.agentHost,
+    port: SymConfigLoader.SymConfig.agentPort,
+    path: `/agent/v4/firehose/${firehoseId}/read`,
+    method: 'GET',
+    headers: {
+      sessionToken: SymBotAuth.sessionAuthToken,
+      keyManagerToken: SymBotAuth.kmAuthToken,
     },
-    'agent': SymConfigLoader.SymConfig.agentProxy
+    agent: SymConfigLoader.SymConfig.agentProxy,
   }
 
-  var req = https.request(options, function (res) {
-    var str = ''
-    res.on('data', function (chunk) {
-      str += chunk
-    })
-    res.on('end', function () {
-      if (SymBotAuth.debug) {
-        console.log('[DEBUG]', 'FirehoseClient/getEventsFromFirehose/res.statusCode', res.statusCode)
-        console.log('[DEBUG]', 'FirehoseClient/getEventsFromFirehouse/str', str)
-      }
-      if (res.statusCode === 200) {
-        PubSub.publish('FIREHOSE: MESSAGE_RECEIVED', SymMessageParser.parse(str))
-        defer.resolve({ 'status': 'success' })
-      } else if (res.statusCode === 204) {
-        defer.resolve({ 'status': 'timeout' })
-      } else {
-        defer.reject({ 'status': 'error' })
-      }
-    })
-  })
-  req.on('error', function (e) {
-    defer.reject({ 'status': 'error' })
+  return request(options, 'FirehoseClient/getEventsFromFirehouse', null, true).then(response => {
+    console.log(
```

Comment:
We are now logging the status code if SymBotAuth.debug is false. Is this intended/ok?

Reply:
👍